### DISCUSSION TITLE
3.4 - Soft Delete Validation Bug?

### DISCUSSION THREAD

**Discussion by Jafo232** [livewire/livewire#6036] [7. Bugs (archived)] [answered]
Here is the method in my component:

```
public function create(): void
    {
        $this->validate([
                            'newName' => 'required|unique:tags,name',
                        ]);

        Tag::create(['name' => $this->newName]);

        $this->dispatch('tagCreated');

    }
```

The tag model uses soft deletes but when I delete a record and then try to add a new record with the same name in this instance, it fails validation with "The new name has already been taken."

I have confirmed the deleted_at column is being set.

![image](https://github.com/livewire/livewire/assets/1456061/ef78a513-1648-4709-982e-6cb65eefbe16)
![image](https://github.com/livewire/livewire/assets/1456061/ad3714bc-6bcf-4c13-8bf8-cc72d724a4f4)
![image](https://github.com/livewire/livewire/assets/1456061/36f08182-6ad4-4127-83a4-d24bb3be8a3f)

**Comment by PhiloNL** [comment, marked as answer]
This is not related to Livewire but to the validation rule. You need to instruct it to ignore soft deleted records.

```php
$this->validate([
  'newName' => ['required', Rule::unique('tags')->withoutTrashed()],
]);
```
